report: github api limit for annotation counts

background:
```bash
2024/07/26 03:59:07.048679 [ERROR] internal/linters/github.go:168: create check run failed: POST https://api.github.com/repos/qbox/stack-hs/check-runs: 422 Invalid request.

No more than 50 items are allowed; 143 were supplied. []
```

Refer: 
```txt
The Checks API limits the number of annotations to a maximum of 50 per API request. To create more than 50 annotations, you have to make multiple requests to the Update a check run endpoint. Each time you update the check run, annotations are appended to the list of annotations that already exist for the check run.
```

--- a/internal/linters/github.go
+++ b/internal/linters/github.go
@@ -137,6 +137,11 @@ func CreateGithubChecks(ctx context.Context, a Agent, lintErrs map[string][]Lint
 	)
 
 	annotations := toGithubCheckRunAnnotations(lintErrs)
+	// limit the number of annotations to 50
+	// see: https://github.com/qiniu/reviewbot/issues/258
+	if len(annotations) > 50 {
+		annotations = annotations[:50]
+	}
 
 	check := github.CreateCheckRunOptions{
 		Name:      linterName,

--- a/internal/runner/runner.go
+++ b/internal/runner/runner.go
@@ -112,7 +112,7 @@ func (*LocalRunner) Run(ctx context.Context, cfg *config.Linter) (io.ReadCloser,
 	return io.NopCloser(bytes.NewReader(output)), execErr
 }
 
-// for easy mock
+// for easy mock.
 type DockerClientInterface interface {
 	ImageInspectWithRaw(ctx context.Context, imageID string) (types.ImageInspect, []byte, error)
 	ImagePull(ctx context.Context, refStr string, options image.PullOptions) (io.ReadCloser, error)

